src/main/java/org/csanchez/aws/glacier/Glacier.java

```
@@ -88,64 +88,77 @@ public Glacier(AWSCredentials credentials, String region) {
+                            cmd.getOptionValue("queue", "glacier"), cmd.getOptionValue("file", "glacier.json"));
+                    break;
+
+                case UPLOAD:
+                    if (arguments.size() < 3) {
+                        throw new GlacierCliException("The upload command requires at least three parameters.");
+                    }
+                    for (String archive : arguments.subList(2, arguments.size())) {
+                        glacier.upload(arguments.get(1), archive);
+                    }
+                    break;
+
+                case DOWNLOAD:
+                    if (arguments.size() != 4) {
+                        throw new GlacierCliException("The download command requires exactly four parameters.");
+                    }
+                    glacier.download(arguments.get(1), arguments.get(2), arguments.get(3));
+                    break;
+
+                case DELETE:
+                    if (arguments.size() != 3) {
+                        throw new GlacierCliException("The delete command requires exactly three parameters.");
+                    }
+                    glacier.delete(arguments.get(1), arguments.get(2));
+                    break;
             }
-            glacier.delete(arguments.get(1), arguments.get(2));
-        } else {
-            printHelp(options);
+        } catch (GlacierCliException e) {
+            System.out.println(e.getClass() + ": " + e.getMessage());
+            System.out.println();
+
+            HelpFormatter formatter = new HelpFormatter();
+            formatter.printHelp("glacier " + "upload vault_name file1 file2 ... | "
+                                           + "download vault_name archiveId output_file | "
+                                           + "delete vault_name archiveId | "
+                                           + "inventory vault_name | ", options);
+        } catch (Exception e) {
```

I don't like a catch-all exception, just let it be thrown for debugging and show the stacktrace